packages/transition-common/src/services/path/Path.ts

```diff
@@ -444,31 +533,66 @@ export class Path extends MapObject<GeoJSON.LineString, PathAttributes> implemen
                 afterNodeWaypointTypes.splice(insertIndex, 0, waypointType);
             }
         }
-        this._attributes.data.waypoints[afterNodeIndex] = afterNodeWaypoints;
-        this._attributes.data.waypointTypes[afterNodeIndex] = afterNodeWaypointTypes;
+        this._attributes.data.waypoints[afterNodeIndex as number] = afterNodeWaypoints;
+        this._attributes.data.waypointTypes[afterNodeIndex as number] = afterNodeWaypointTypes;
         this._updateHistory();
         return this.updateGeography();
     }
 
+    /**
+     * Replace waypoint coordinates by new ones. If the waypoint does not exist,
+     * the path is not changed.
+     *
+     * @param waypointCoordinates The new coordinates of the waypoint
+     * @param waypointType The type of this updated waypoint
+     * @param afterNodeIndex The index of the node after which the waypoint is
+     * located
+     * @param waypointIndex The index of the waypoint to update
+     * @returns The updated path
+     */
     updateWaypoint(
         waypointCoordinates: [number, number],
-        waypointType = null,
+        waypointType: string | undefined = undefined,
```

Comment:
Should we add a type here?

Reply:
We need to add type for the return value ideally 2 lines below, do you mean here in this line? there is a type...

Reply:
I mean should we replace string by possible values?

Reply:
Indeed, we eventually should, but we can wait after the refactoring and maybe profile congestion to see if 'manual' | 'engine' is the complete and right type.